src/PrestaShopBundle/Translation/Provider/ExternalModuleLegacySystemProvider.php

```diff
@@ -0,0 +1,216 @@
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Translation\Provider;
+
+use PrestaShopBundle\Translation\Exception\UnsupportedLocaleException;
+use PrestaShopBundle\Translation\Extractor\LegacyModuleExtractorInterface;
+use Symfony\Component\Translation\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * Be able to retrieve information from legacy translation files
+ */
+class ExternalModuleLegacySystemProvider extends AbstractProvider implements UseDefaultCatalogueInterface, SearchProviderInterface
+{
+    /**
+     * @var SearchProviderInterface the module provider
+     */
+    private $moduleProvider;
+
+    /**
+     * @var LoaderInterface the translation loader from legacy files
+     */
+    private $legacyFileLoader;
+
+    /**
+     * @var LegacyModuleExtractorInterface the extractor
+     */
+    private $legacyModuleExtractor;
+
+    /**
+     * @var string the module name
+     */
+    private $moduleName;
+
+    public function __construct(
+        LoaderInterface $databaseLoader,
+        $resourceDirectory,
+        LoaderInterface $legacyFileLoader,
+        LegacyModuleExtractorInterface $legacyModuleExtractor,
+        SearchProviderInterface $moduleProvider
+    ) {
+        $this->moduleProvider = $moduleProvider;
+        $this->legacyFileLoader = $legacyFileLoader;
+        $this->legacyModuleExtractor = $legacyModuleExtractor;
+
+        parent::__construct($databaseLoader, $resourceDirectory);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTranslationDomains()
+    {
+        return ['#^' . $this->getModuleDomain() . '*#i'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIdentifier()
+    {
+        return 'external_legacy_module';
+    }
+
+    /**
+     * is/should module name be part of the API?
+     */
+    public function setModuleName($moduleName)
+    {
+        $this->moduleName = $moduleName;
+
+        return $this;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultCatalogue($empty = true)
+    {
+        $defaultCatalogue = $this->moduleProvider
+            ->setLocale($this->locale)
+            ->getDefaultCatalogue()
+        ;
+
+        try {
+            $additionalDefaultCatalogue = $this->legacyModuleExtractor->extract($this->moduleName, $this->locale);
+            $defaultCatalogue->addCatalogue($additionalDefaultCatalogue);
+        } catch (UnsupportedLocaleException $exception) {
+            // Do nothing as support of legacy file is deprecated
+        }
+
+        if ($empty) {
+            $defaultCatalogue = $this->emptyCatalogue($defaultCatalogue);
+        }
+
+        return $defaultCatalogue;
+    }
+
+    public function getXliffCatalogue()
+    {
+        return new MessageCatalogue($this->locale);
```

Comment:
It's weird to generate a new catalog each time you call getXliffCatalogue.

Reply:
I won't introduce a cache system in this pull request, but I agree with you: a LOT of improvements can be done in the translations workflow :+1:

Reply:
I can't imagine, this is why we must do our best to not introduce something like that if your future pull requests will be merged in 2020 :trollface: